base_layer/wallet_ffi/src/error.rs

```diff
@@ -213,6 +213,12 @@ impl From<WalletError> for LibWalletError {
                 code: 204,
                 message: format!("{:?}", w),
             },
+            WalletError::TransactionServiceError(TransactionServiceError::OutputManagerError(
+                OutputManagerError::BuildError(ref s),
+            )) if s == &"Fee is greater than amount".to_string() => Self {
```

Comment:
Not to be a pain, but this should probably be a const used in both places. Otherwise someone might change the other one without updating this one.

Reply:
Would require a lot of const's, the Build Error is all strings, worst case here it will be caught by error 204

Reply:
Ideally, BuildError should be an enum but not really looking to refactor it here

Reply:
Sorry, I just meant making "Fee is greater than amount" a const &str and used here, and also the place where it's return (transaction init?) 

```rust
pub const FEE_GT_AMOUNT_ERR: &str = "Fee is greater than amount";
```

Since you're relying on the strings being the same in both places

Reply:
I would say that using const's is not the way to clean this up. It should be to refactor the TransactionError::BuildError to use the normal Anyhow Error paradigm we use everywhere else:

https://github.com/tari-project/tari/blob/development/base_layer/core/src/transactions/transaction_protocol/transaction_initializer.rs#L611

But that is not a very small job so happy to put it in the backlog.

Reply:
That's fine I agree, the point was that if this string is changed there it will break your check here

Reply:
Ja I agree it's not ideal and have made a ticket to refactor BuildError based on the fragility this issue has revealed. Just didn't want to normalize the const strategy in the code base when the canonical solution should be to use Error enums.